app/components/Calendar/index.js

```diff
@@ -0,0 +1,442 @@
+      left: '', // 'myPrev,myNext,myToday',
+      center: 'timeGridDay,timeGridWeek',
+      right: '',
+    },
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    const { date } = this.props;
+    const { viewType } = this.state;
+
+    const calendarApi = this.calendarRef.current.getApi();
+
+    if (date !== prevProps.date) {
+      calendarApi.gotoDate(date);
+    }
+    if (viewType !== prevState.viewType) {
+      calendarApi.changeView(viewType);
+    }
+  }
+
+  isSelectionValid = selection => {
+    const { resource } = this.props;
+    const events = this.getReservedEvents();
+
+    return isTimeRangeReservable(
+      resource,
+      selection.start,
+      selection.end,
+      events,
+    );
+  };
+
+  onChange = selected => {
+    this.setState({ selected });
+
+    // Invoke select handler callback from props
+    this.props.onTimeChange(selected);
+  };
+
+  onCancel = () => {
+    // Revert to default timerange if cancel
+    const defaultSelectedTimeRange = calendarUtils.getDefaultSelectedTimeRange(
+      this.props.editingReservation,
+    );
+    this.onChange(defaultSelectedTimeRange);
+  };
+
+  onEventRender = info => {
+    // add cancel button for new selected event
+    let duration;
+
+    if (info.event.id === NEW_RESERVATION) {
+      const cancelBtn = document.createElement('span');
+      cancelBtn.classList.add('app-TimePickerCalendar__cancelEvent');
+      cancelBtn.addEventListener('click', () => this.onCancel(), {
+        once: true,
+      });
+      info.el.append(cancelBtn);
+      duration = this.getDurationText(info.event);
+    } else if (info.event.id === '') {
+      duration = this.getDurationText(info.event);
+    }
+
+    if (duration) {
+      const eventDuration = document.createElement('span');
+      eventDuration.textContent = duration;
+      eventDuration.classList.add('app-TimePickerCalendar__maxDuration');
+      info.el.append(eventDuration);
+    }
+  };
+
+  onSelect = selectionInfo => {
+    // const { t } = this.props;
+    const calendarApi = this.calendarRef.current.getApi();
+    calendarApi.unselect();
+    // Clear FullCalendar select tooltip
+
+    const selectable = this.getSelectableTimeRange(selectionInfo);
+    // Make sure selected time alway furfill period check.
+
+    const isValid = this.isSelectionValid(selectable);
+
+    if (isValid) {
+      this.onChange(selectable);
+    } else {
+      // TODO: Display error
+    }
+  };
+
+  // Check if event resize allowed
+  onEventResize = selectionInfo => {
+    const { event } = selectionInfo;
+    const selectable = this.getSelectableTimeRange(event, selectionInfo);
+    this.onChange(selectable);
+  };
+
+  onDatesRender = info => {
+    const { viewType, header } = this.state;
+    let view = info.view.type;
+    let headerConfig = header;
+
+    // 768
+    if (window.innerWidth < 768) {
+      // mobile view config
+      view = 'timeGridDay';
+      headerConfig = {
+        left: 'myPrev,myNext,myToday',
```

Comment:
These buttons are commented out on initial state. If it's intended to never show these, I suggest you comment these out as well. Even though tablet in portrait mode will probably have more width than 768, you never know what users manage to achieve. :smile: